ios/keyman/Keyman/Keyman/DropDownList/DropDownListView.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  DropDownListView.swift
+//  Keyman
+//
+//  Created by Gabriel Wong on 2017-09-07.
+//  Copyright © 2017 SIL International. All rights reserved.
+//
+
+import UIKit
+
+class DropDownListView: UIView {
+  init(listItems items: [UIBarButtonItem], itemSize size: CGSize, position pos: CGPoint) {
+    let frameHeight = items.count > 0 ? size.height * CGFloat(items.count) : size.height
+    let frame = CGRect(x: pos.x, y: pos.y, width: size.width, height: frameHeight)
+    super.init(frame: frame)
+
+    backgroundColor = UIColor.clear
+    let dropDownView = DropDownView(frame: CGRect.zero)
+    let arrowHeight = dropDownView.arrowHeight
+    dropDownView.frame = CGRect(x: 0, y: -arrowHeight, width: frame.size.width,
+        height: frame.size.height + arrowHeight)
+    dropDownView.arrowPosX = frame.size.width - 29
+    addSubview(dropDownView)
+
+    let count: Int = items.count
+    let w: CGFloat = frame.width
+    let h: CGFloat = size.height
+    let x: CGFloat = 0
+    var y: CGFloat = 0
+
+    for (index, item) in items.enumerated() {
+      let button = UIButton(type: .custom)
+      button.setTitleColor(UIColor.lightGray, for: .normal)
+      button.setTitleColor(UIColor.darkGray, for: .highlighted)
+      button.frame = CGRect(x: x, y: y, width: w, height: h)
+      button.setTitle(item.title, for: .normal)
+      addSubview(button)
+      if let subviews = item.customView?.subviews, subviews.count > 0,
+          let btn = subviews[0] as? UIButton {
```

Comment:
`btn` did not used to be part of the condition; is there a reason you wrote it that way?

````objc
     NSArray *subViews = [[item customView] subviews];
     if ([subViews count]) {
         UIButton *btn = [[[item customView] subviews] objectAtIndex:0];
````

Reply:
It's because `subviews` has type `[UIView]` so I put it in the condition to safely cast to `UIButton`. The Obj-C code casts in an unsafe way.